dds/DCPS/XTypes/DynamicData.cpp

```diff
@@ -2193,6 +2239,458 @@ const char* DynamicData::typekind_to_string(TypeKind tk) const
+    break;
+  }
+  // case OpenDDS::XTypes::TK_BITMASK:
+  // case OpenDDS::XTypes::TK_BITSET:
+  case OpenDDS::XTypes::TK_ALIAS: {
+    if (!print_dynamic_data(dd, type_string, indent)) {
+      if (OpenDDS::DCPS::log_level >= OpenDDS::DCPS::LogLevel::Error) {
+        ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: print_dynamic_data: failed to read alias\n"));
+      }
+      return false;
+    }
+    break;
+  }
+  case OpenDDS::XTypes::TK_SEQUENCE: {
+    OpenDDS::DCPS::String temp_indent = indent;
+    indent += "  ";
+    type_name = dd.type()->get_descriptor().element_type->get_descriptor().name;
+    member_name = dd.type()->get_descriptor().name;
+    ACE_CDR::ULong seq_length = dd.get_item_count();
+    type_string += "  " + type_name + "[" + OpenDDS::DCPS::to_dds_string(seq_length) + "] " +  member_name + " =\n";
+    for (ACE_CDR::ULong i = 0; i < seq_length; ++i) {
+      type_string += indent + "[" + OpenDDS::DCPS::to_dds_string(i) + "]";
+      if (dd.get_complex_value(temp_dd, i) != DDS::RETCODE_OK) {
+        if (OpenDDS::DCPS::log_level >= OpenDDS::DCPS::LogLevel::Error) {
+          ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: print_dynamic_data: failed to get_complex_value\n"));
+        }
+        return false;
+      }
+      if (!print_dynamic_data(temp_dd, type_string, indent)) {
+        if (OpenDDS::DCPS::log_level >= OpenDDS::DCPS::LogLevel::Error) {
+          ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: print_dynamic_data: failed to read struct member\n"));
+        }
+        return false;
+      }
+    }
+    indent = temp_indent;
+    break;
+  }
+  case OpenDDS::XTypes::TK_ARRAY: {
+    OpenDDS::DCPS::String temp_indent = indent;
+    indent += "  ";
+    type_name = dd.type()->get_descriptor().element_type->get_descriptor().name;
+    member_name = dd.type()->get_descriptor().name;
+    OpenDDS::XTypes::LBound bound = dd.type()->get_descriptor().bound[0];
+    type_string += "  " + type_name + "[" + OpenDDS::DCPS::to_dds_string(bound) + "] " +  member_name + " =\n";
+    for (ACE_CDR::ULong i = 0; i < bound; ++i) {
+      type_string += indent + "[" + OpenDDS::DCPS::to_dds_string(i) + "]";
+      if (dd.get_complex_value(temp_dd, i) != DDS::RETCODE_OK) {
+        if (OpenDDS::DCPS::log_level >= OpenDDS::DCPS::LogLevel::Error) {
+          ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: print_dynamic_data: failed to get_complex_value\n"));
+        }
+        return false;
+      }
+      if (!print_dynamic_data(temp_dd, type_string, indent)) {
+        if (OpenDDS::DCPS::log_level >= OpenDDS::DCPS::LogLevel::Error) {
+          ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: print_dynamic_data: failed to read struct member\n"));
+        }
+        return false;
+      }
+    }
+    indent = temp_indent;
+    break;
+  }
+  case OpenDDS::XTypes::TK_STRUCTURE: {
+    OpenDDS::DCPS::String temp_indent = indent;
+    indent += "  ";
+    type_string += "struct " + dd.type()->get_name() + " {\n";
```

Comment:
This could be up to debate, but I'm not sure we should mimic IDL necessarily.

Reply:
Is there an example from the sample dissector that would be helpful here?

Reply:
I suppose. This is a screenshot of what that looks like from my article: ![](https://objectcomputing.com/files/cache/9698b0dd3f894c844594f8544d82b830_f5327.png)

Another example of non-JSON text representation would be the `Printer` though, it's basically the same. Here I added the printer to DevGuideMessenger along with a nested struct because I didn't remember how it handled that.

```
    from: "Comic Book Guy"
    subject: "Review"
    subject_id: 107
    text: "Worst. Movie. Ever."
    count: 8
    n:
        value: 2147483647
```

This is probably the first time this code has ran since I worked on it so I'm glad to see it still works. I don't have an issue with printing out the type name ([`Printer` has a todo for this](https://github.com/objectcomputing/OpenDDS/blob/05c82664fc481a2d9cc620d3eb4f00f9911c14ff/dds/DCPS/Printer.h#L41)) or things like `struct`. I do thing we could omit the braces though as the indent should be enough.